File: task2/range.py
def generate_range(min, max, step):
    return list(range(min, max + 1, step))


print(generate_range(2, 10, 2))  # should return list of [2,4,6,8,10]
print(generate_range(1, 10, 3))  # should return list of [1,4,7,10]
print(generate_range(2, 10, 2))  # should return array of [2, 4, 6, 8, 10]
print(generate_range(1, 10, 3))  # should return array of [1, 4, 7, 10]

File: archive/task02/range.c
#include <stdio.h>
#include <stdlib.h>

int *generate_range(int min, int max, int step, size_t *sz) {
    int* result = (int*) malloc(*sz * sizeof (int));

    for (int i = 0; i < *sz; ++i) {
        result[i] = min;
        min += step;
    }

    return result;
}

int main(void) {
    size_t sz = 4;

    int* res = generate_range(1, 10, 3, &sz);

    for (int i = 0; i < sz; ++i) {
        printf("%d\n", res[i]);
    }

    return 0;
}
